src/checkbox/checkbox.js

```diff
@@ -15,255 +15,182 @@
+   * @export
    */
-  MaterialCheckbox.prototype.blur_ = function() {
-    // TODO: figure out why there's a focus event being fired after our blur,
-    // so that we can avoid this hack.
-    window.setTimeout(function() {
-      this.inputElement_.blur();
-    }.bind(this), /** @type {number} */ (this.Constant_.TINY_TIMEOUT));
-  };
-
-  // Public methods.
+  set checked(value) {
+    this.input_.checked = value;
+    this.refresh();
+  }
 
   /**
-   * Check the inputs toggle state and update display.
+   * Return "checked" value on checkbox.
    *
-   * @public
+   * @return {boolean} The current value of the property.
+   * @export
    */
-  MaterialCheckbox.prototype.checkToggleState = function() {
-    if (this.inputElement_.checked) {
-      this.element_.classList.add(this.CssClasses_.IS_CHECKED);
-    } else {
-      this.element_.classList.remove(this.CssClasses_.IS_CHECKED);
-    }
-  };
-  MaterialCheckbox.prototype['checkToggleState'] =
-      MaterialCheckbox.prototype.checkToggleState;
+  get checked() {
+    return this.input_.checked;
+  }
 
   /**
-   * Check the inputs disabled state and update display.
+   * Disable / enable the checkbox component.
    *
-   * @public
+   * @param {boolean} value The value to set the property to.
+   * @export
    */
-  MaterialCheckbox.prototype.checkDisabled = function() {
-    if (this.inputElement_.disabled) {
-      this.element_.classList.add(this.CssClasses_.IS_DISABLED);
-    } else {
-      this.element_.classList.remove(this.CssClasses_.IS_DISABLED);
-    }
-  };
-  MaterialCheckbox.prototype['checkDisabled'] =
-      MaterialCheckbox.prototype.checkDisabled;
+  set disabled(value) {
+    this.input_.disabled = value;
+    this.refresh();
+  }
 
   /**
-   * Disable checkbox.
+   * Return whether the checkbox component is disabled or enabled.
    *
-   * @public
+   * @return {boolean} The current value of the property.
+   * @export
    */
-  MaterialCheckbox.prototype.disable = function() {
-    this.inputElement_.disabled = true;
-    this.updateClasses_();
-  };
-  MaterialCheckbox.prototype['disable'] = MaterialCheckbox.prototype.disable;
+  get disabled() {
+    return this.input_.disabled;
+  }
 
   /**
-   * Enable checkbox.
+   * Run a visual refresh on the component, in case it's gone out of sync.
    *
-   * @public
+   * @export
    */
-  MaterialCheckbox.prototype.enable = function() {
-    this.inputElement_.disabled = false;
-    this.updateClasses_();
-  };
-  MaterialCheckbox.prototype['enable'] = MaterialCheckbox.prototype.enable;
+  refresh() {
+    this.checkDisabled_();
+    this.checkToggleState_();
+  }
 
   /**
-   * Check checkbox.
+   * Add blur.
    *
-   * @public
+   * @private
    */
-  MaterialCheckbox.prototype.check = function() {
-    this.inputElement_.checked = true;
-    this.updateClasses_();
-  };
-  MaterialCheckbox.prototype['check'] = MaterialCheckbox.prototype.check;
+  blur_() {
+    requestAnimationFrame(() => this.input_.blur());
```

Comment:
That results in an `Uncaught TypeError: Illegal invocation`, presumably because of context issues.